webui/dev/index.tsx

```diff
@@ -16,41 +16,64 @@ import { createMuiTheme } from '@material-ui/core';
 import { Main } from '../src/main';
 import { ExtensionRegistryService } from '../src/extension-registry-service';
 import createPageSettings from './page-settings';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
 
 // This file is the main entry point for the development setup.
 // The production code is at https://github.com/eclipse/open-vsx.org
 
-const theme = createMuiTheme({
-    palette: {
-        primary: { main: '#eeeeee', contrastText: '#3f3841', dark: '#565157' },
-        secondary: { main: '#a60ee5', contrastText: '#edf5ea' }
-    },
-    breakpoints: {
-        values: {
-            xs: 340,
-            sm: 550,
-            md: 800,
-            lg: 1040,
-            xl: 1240
-        }
-    }
-});
-
 let serverHost = location.hostname;
 if (serverHost.startsWith('3000-')) {
     // The frontend runs on port 3000, but the server runs on port 8080
     serverHost = '8080-' + serverHost.substring(5);
 }
-const service = new ExtensionRegistryService(`${location.protocol}//${serverHost}`);
 
-const pageSettings = createPageSettings(theme);
+const App = () => {
+    const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+
+    const theme = React.useMemo(
+        () =>
+            createMuiTheme({
+                palette: {
+                    primary: {
+                        main: prefersDarkMode ? '#444' : '#eeeeee',
+                        dark: prefersDarkMode ? '#f4f4f4' : '#565157'
+                    },
+                    secondary: {
+                        main: prefersDarkMode ? '#c160ef' : '#a60ee5', 
+                        contrastText: '#edf5ea',
+                        light: prefersDarkMode ? '#777' : '#fff',
+                        dark: prefersDarkMode ? '#333' : '#fff', 
+                    },
+                    type: prefersDarkMode ? 'dark' : 'light'
+                },
+                breakpoints: {
+                    values: {
+                        xs: 340,
+                        sm: 550,
+                        md: 800,
+                        lg: 1040,
+                        xl: 1240
+                    }
+                }
+            }),
+        [prefersDarkMode],
+    );
+
+    const service = new ExtensionRegistryService(`${location.protocol}//${serverHost}`);
+    const pageSettings = createPageSettings(theme, prefersDarkMode);
+
+    return (
+        <ThemeProvider theme={theme}>
+            <Main
+                service={service}
+                pageSettings={pageSettings}
+                prefersDarkMode={prefersDarkMode}
```

Comment:
You could include a property `darkTheme` in the PageSettings so you don't have to pass this additional boolean through the components.

Reply:
Or maybe even better: a property `themeType` of type `'light' | 'dark`, then you can combine it more easily with the `extension.galleryTheme` property (see below).